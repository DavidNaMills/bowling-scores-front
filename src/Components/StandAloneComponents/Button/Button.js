import React from 'react';
import classes from './Button.module.scss';



const Button = ({ click = () => { }, label, isDisabled = false, type = 'default' }) => {
    let style = [classes.button];

    if (isDisabled) {
        style = style.concat(classes.button__disabled);
    } else {

        switch (type.toLowerCase()) {
            case 'warning':
                style = style.concat(classes.button__warning);
                break;
            case 'danger':
                style = style.concat(classes.button__danger);
                break;
            default:
                style = style.concat(classes.button__default);
                break;
        }
    }

    return (
        <button onClick={click} disabled={isDisabled} className={style.join(' ')}>{label}</button>
    );
};

export default Button;