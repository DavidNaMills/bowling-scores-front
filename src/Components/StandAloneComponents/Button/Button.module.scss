@import '../../../styles/colors.scss';


.button{
    border: none;
    font-size: 16px;
    padding: 10px;
    min-width: 100px;
    font-weight: 550;

    &__disabled{
        background-color: $btn_disabled;
    }
    
    &__isFull{
        width: 100%;
    }

    &__default{
        background-color: $btn_default_bkg;
        color: $btn_font_clr_light;
        
        &:hover{
            background-color: $btn_default_bkg_active_hover;
        }
        &:active{
            background-color: $btn_default_bkg_active;
        }
    }

    &__warning{
        background-color: $btn_warning_bkg;
        color: $btn_font_clr;
        
        &:hover{
            background-color: $btn_warning_bkg_active_hover;
        }
        &:active{
            background-color: $btn_warning_bkg_active;
        }    
    }
    
    &__danger{
        background-color: $btn_danger_bkg;
        color: $btn_font_clr_light;
        
        &:hover{
            background-color: $btn_danger_bkg_active_hover;
        }
        &:active{
            background-color: $btn_danger_bkg_active;
        }    
    }

    &__lightGreen{
        background-color: $btn_lightGreen_bkg;
        color: $btn_font_clr;
        
        &:hover{
            background-color: $btn_lightGreen_bkg_active;
        }
        &:active{
            background-color: $btn_lightGreen_bkg_active_hover;
        }    
    }

    &__darkGreen{
        background-color: $btn_darkGreen_bkg;
        color: $btn_font_clr_light;
        
        &:hover{
            background-color: $btn_darkGreen_bkg_active;
        }
        &:active{
            background-color: $btn_darkGreen_bkg_active_hover;
        }    
    }

    &__blue{
        background-color: $btn_blue_bkg;
        color: $btn_font_clr_light;
        
        &:hover{
            background-color: $btn_blue_bkg_active;
        }
        &:active{
            background-color: $btn_blue_bkg_active_hover;
        }    
    }

    &__lightBlue{
        background-color: $btn_lightBlue_bkg;
        color: $btn_font_clr;
        
        &:hover{
            background-color: $btn_lightBlue_bkg_active;
        }
        &:active{
            background-color: $btn_lightBlue_bkg_active_hover;
        }    
    }

    &__darkBlue{
        background-color: $btn_darkBlue_bkg;
        color: $btn_font_clr_light;
        
        &:hover{
            background-color: $btn_darkBlue_bkg_active;
        }
        &:active{
            background-color: $btn_darkBlue_bkg_active_hover;
        }    
    }

    &__lightRed{
        background-color: $btn_lightRed_bkg;
        color: $btn_font_clr;
        
        &:hover{
            color: $btn_font_clr_light;
            background-color: $btn_lightRed_bkg_active;
        }
        &:active{
            background-color: $btn_lightRed_bkg_active_hover;
        }    
    }

    &__lightYellow{
        background-color: $btn_lightYellow_bkg;
        color: $btn_font_clr;
        
        &:hover{
            color: $btn_font_clr_light;
            background-color: $btn_lightYellow_bkg_active;
        }
        &:active{
            background-color: $btn_lightYellow_bkg_active_hover;
        }    
    }



}