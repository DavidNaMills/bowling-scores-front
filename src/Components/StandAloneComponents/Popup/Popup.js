import React from 'react';
import classes from '../../../styles/shared/Popup.module.scss';

import useBodyLock from '../../../Hooks/useBodyLock/useBodyLock';

const Popup = ({ close = null, children }) => {
    useBodyLock();
    return (
        <div className={classes.popup__container}>
            <div className={classes.popup__module}>
                {close &&
                    <div className={classes.popup__closeBar}>
                        <p onClick={close} className={classes.popup__closeBar_cls}>CLOSE</p>
                    </div>
                }
                <div>
                    {children}
                </div>
            </div>
        </div>
    )
}

export default Popup;