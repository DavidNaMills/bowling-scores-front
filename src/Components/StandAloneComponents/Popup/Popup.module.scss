@import '../../../styles/colors.scss';

.popup{
    &__container{
        position: absolute;
        top:0;
        left: 0;
        background-color: $popup_container_color;
        height: 100vh;
        width: 100vw;
        z-index: 900;
        display: flex;
        justify-content: center;
        align-items: center;
    }

    &__module{
        border: $popup_borders;
        background: $popup_module_color;
        margin: auto;
        padding: 10px;
        width: 80vw;
        max-width: 400px;
    }

    &__closeBar{
        padding:0;
        display: flex;
        justify-content: left;
        
        &_cls{
            font-size: 12px;
            color: $btn_danger_bkg_active_hover;
        }
    }
    
    &__message{
        padding: 30px;
        border-bottom: $popup_borders_separator;
    }

    &__btnContainer{
        display: flex;
        justify-content: space-around;
        padding-top: 10px;
        margin: 20px;
    }

}