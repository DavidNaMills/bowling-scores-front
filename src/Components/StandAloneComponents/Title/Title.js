import React from 'react';
import classes from './Title.module.scss';

const Title = ({ label = '', ttlType = 'sub' }) => {
    let item;

    switch (ttlType.toLowerCase()) {
        case 'main':
            item = <h1 className={classes.title__main}>{label}</h1>
            break;
        case 'section':
            item = <h2 className={classes.title__section}>{label}</h2>
            break;
        case 'sub':
            item = <h3 className={classes.title__sub}>{label}</h3>
            break;
        default:
            item = <h3 className={classes.title__sub}>{label}</h3>
            break;
    }

    return (
        <div className={classes.title}>
            {item}
        </div>
    )
}

export default Title;