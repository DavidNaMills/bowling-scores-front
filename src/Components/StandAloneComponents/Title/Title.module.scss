@import '../../../styles/colors.scss';
@import '../../../styles/fonts.scss';

.title{
    // -webkit-box-shadow: $title_shadow_full;
    // -moz-box-shadow: $title_shadow_full;
    // box-shadow: $title_shadow_full;
    padding: 2px;
    text-align: center;
    font-family: $header;

    &__main{
        font-size: $header_size_main;
        color: $title_main_color;
    }
    
    &__section{
        font-size: $header_size_section;
        color: $title_rest_color;
    }
    &__sub{
        font-size: $header_size_sub;
        color: $title_rest_color;
    }
}