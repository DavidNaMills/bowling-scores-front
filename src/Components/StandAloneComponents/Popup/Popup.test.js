import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import Popup from './Popup';
import Button from '../Button/Button';
import Title from '../Title/Title';


const buttonTest = [
    {
        msg: 'Doesnt render any buttons',
        test: {},
        results: {
            count: 0
        }
    },
    {
        msg: 'Only creates 1 button through action1',
        test: {
            action1: {
                action: jest.fn(),
                label: 'button 1'
            }
        },
        results: {
            count: 1
        }
    },
    {
        msg: 'Only renders 1 button through action2',
        test: {
            action2: {
                action: jest.fn(),
                label: 'button 2'
            }
        },
        results: {
            count: 1
        }
    },
    {
        msg: 'renders 2 buttons',
        test: {
            action1: {
                action: jest.fn(),
                label: 'button 1'
            },
            action2: {
                action: jest.fn(),
                label: 'button 2',
                type: 'danger'
            },
        },
        results: {
            count: 2
        }
    },

]

describe('<Popup /> test suite', () => {
    describe('snapshot test', ()=>{
        test('matches default snapshot', () => {
            const testMesg = 'This is a test message';
            const component = renderer.create(<Popup message={testMesg} />);
            const app = component.toJSON();
            expect(app).toMatchSnapshot();
        });
    });

    describe('styling test suite', () => {
        it('displays the message with no optional extras', () => {
            const testMesg = 'This is a test message';
            const wrapper = mount(<Popup message={testMesg} />);
            expect(wrapper.find('p').length).toBe(1);
            expect(wrapper.find('p').at(0).text()).toEqual(testMesg);
            expect(wrapper.find(Title).length).toBe(0);
            expect(wrapper.find(Button).length).toBe(0);
        });

        it('contains a counter className and a module className', () => {
            const wrapper = mount(<Popup />)
            expect(wrapper.find('.popup__container').length).toBe(1);
            expect(wrapper.find('.popup__module').length).toBe(1);
        });


        buttonTest.forEach(x => {
            it(x.msg, () => {
                const wrapper = mount(<Popup {...x.test} />)
                const btn = wrapper.find(Button);
                expect(btn.length).toBe(x.results.count);
            });
        });

        it('contains a message', () => {
            const testMsg = 'This is a test message';
            const wrapper = mount(<Popup message={testMsg} />)
            const msg = wrapper.find('p');
            expect(msg.at(0).length).toBe(1);
            expect(msg.at(0).text()).toEqual(testMsg);
            expect(wrapper.find('.popup__message').length).toBe(1);
        });

        it('contains a title when specified', () => {
            const testTitle = {
                label: 'Test title',
                ttlType: 'section'
            }
            const wrapper = mount(<Popup title={testTitle} />);
            expect(wrapper.find(Title).length).toBe(1);
        });

        it('contains a <p> tag with text CLOSE ', () => {
            const wrapper = mount(<Popup close={()=>{}}/>);
            const cls = wrapper.find('p');
            expect(cls.length).toBe(2);
            expect(cls.at(0).text()).toEqual('CLOSE');
            expect(wrapper.find('.popup__closeBar').length).toBe(1);
            expect(wrapper.find('.popup__closeBar_cls').length).toBe(1);
        });
    
        it(' CLOSE is not rendered if no close function is provided ', () => {
            const wrapper = mount(<Popup />);
            const cls = wrapper.find('p');
            expect(cls.length).toBe(1);
        });
    });

    describe('behavioural test suite', () => {
        it('fires the close function', () => {
            const closeMk = jest.fn();
            const wrapper = mount(<Popup close={closeMk} />);
            wrapper.find('p').at(0).simulate('click');
            expect(closeMk.mock.calls.length).toBe(1);
        });
    });
});
