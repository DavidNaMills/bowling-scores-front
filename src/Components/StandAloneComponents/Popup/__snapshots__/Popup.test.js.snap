// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Popup /> test suite snapshot test matches default snapshot 1`] = `
<div
  className="popup__container"
>
  <div
    className="popup__module"
  >
    <div>
      <p
        className="popup__message2"
      >
        This is a test message
      </p>
      <div
        className="popup__btnContainer"
      />
    </div>
  </div>
</div>
`;
